e2e/tests/omnibus-base-mainnet-andromeda.toml/Perps_Trading.e2e.js

```diff
@@ -0,0 +1,308 @@
+const { setSettlementDelays } = require('../../tasks/setPerpsSettlementDelays');
+const { getPerpsSettlementStrategy } = require('../../tasks/getPerpsSettlementStrategy');
+
+const SpotMarketProxyDeployment = require('../../deployments/SpotMarketProxy.json');
+const PerpsMarketProxyDeployment = require('../../deployments/PerpsMarketProxy.json');
+const extras = require('../../deployments/extras.json');
+
+const log = require('debug')(`e2e:${require('path').basename(__filename, '.e2e.js')}`);
+
+const wait = (ms) =>
+  new Promise((resolve) => {
+    require('debug')(`e2e:wait`)('Start', { ms });
+    setTimeout(() => {
+      require('debug')(`e2e:wait`)('Finish', { ms });
+      resolve();
+    }, ms);
+  });
+
+describe(require('path').basename(__filename, '.e2e.js'), function () {
+  const accountId = parseInt(`420${crypto.randomInt(1000)}`);
+  const provider = new ethers.providers.JsonRpcProvider(
+    process.env.RPC_URL || 'http://127.0.0.1:8545'
+  );
+  const wallet = ethers.Wallet.createRandom().connect(provider);
+  const address = wallet.address;
+  const privateKey = wallet.privateKey;
+
+  const PerpsMarketProxy = new ethers.Contract(
+    PerpsMarketProxyDeployment.address,
+    PerpsMarketProxyDeployment.abi,
+    wallet
+  );
+
+  it('should create new random wallet', async () => {
+    log({ wallet: wallet.address, pk: wallet.privateKey });
+    assert.ok(wallet.address);
+  });
+
+  it('should set ETH balance to 100', async () => {
+    const address = wallet.address;
+    assert.equal(await getEthBalance({ address }), 0, 'New wallet has 0 ETH balance');
+    await setEthBalance({ address, balance: 100 });
+    assert.equal(await getEthBalance({ address }), 100);
+  });
+
+  it('should set USDC balance to 100_000', async () => {
+    assert.equal(
+      await getCollateralBalance({ address, symbol: 'USDC' }),
+      0,
+      'New wallet has 0 USDC balance'
+    );
+    await setUSDCTokenBalance({ wallet, balance: 100_000 });
+    assert.equal(await getCollateralBalance({ address, symbol: 'USDC' }), 100_000);
+  });
+
+  it('should approve USDC spending for SpotMarket', async () => {
+    assert.equal(
+      await isCollateralApproved({
+        address,
+        symbol: 'USDC',
+        spenderAddress: SpotMarketProxyDeployment.address,
+      }),
+      false,
+      'New wallet has not allowed SpotMarket USDC spending'
+    );
+    await approveCollateral({
+      privateKey,
+      symbol: 'USDC',
+      spenderAddress: SpotMarketProxyDeployment.address,
+    });
+    assert.equal(
+      await isCollateralApproved({
+        address,
+        symbol: 'USDC',
+        spenderAddress: SpotMarketProxyDeployment.address,
+      }),
+      true
+    );
+  });
+
+  it('should make a price update', async () => {
+    // commitOrder and views requiring price will fail if there's no price update within the last hour,
+    // so we send off a price update just to be safe
+    await Promise.all([
```

Comment:
I think this needs to be sequential, otherwise we'll get nonce already used

Reply:
yep already fixed =(